packages/gatsby/src/redux/index.js

```diff
@@ -32,6 +32,7 @@ try {
     initialState.staticQueryComponents
   )
   initialState.components = objectToMap(initialState.components)
+  initialState.nodes = objectToMap(initialState.nodes)
```

Comment:
I think we need to do also reverse operation when we serialize data before saving to `redux-state.json`:

https://github.com/gatsbyjs/gatsby/blob/6a8e9a05ac95a41f5bc02d002bc73be4dec6a604/packages/gatsby/src/redux/index.js#L85-L93

At least in my tests it would store nodes Map as empty object and would need to run queries again, because all nodes would be new.

Reply:
Fixed, thanks!